app/form_models/application_details_form.rb

```diff
@@ -1,15 +1,81 @@
 class ApplicationDetailsForm < VacancyForm
+  validate :validate_expiry_time
+
   delegate :expires_on_dd, :expires_on_mm, :expires_on_yyyy,
            :publish_on_dd, :publish_on_mm, :publish_on_yyyy,
-           :published?, :status, :publish_on_changed?, to: :vacancy
+           :published?, :status, :publish_on_changed?, :expiry_time, to: :vacancy
+
+  attr_accessor :params, :expiry_time_hh, :expiry_time_mm, :expiry_time_meridian
 
   include VacancyApplicationDetailValidations
 
+  def initialize(params)
+    @params = params
+    @expiry_time_hh = params.delete(:expiry_time_hh) || params[:expiry_time]&.strftime('%-l')
+    @expiry_time_mm = params.delete(:expiry_time_mm) || params[:expiry_time]&.strftime('%-M')
+    @expiry_time_meridian = params.delete(:expiry_time_meridian) || params[:expiry_time]&.strftime('%P')
+
+    super(params)
+  end
+
   def disable_editing_publish_on?
     published? && vacancy.reload.publish_on.past?
   end
 
   def completed?
-    application_link && contact_email && publish_on && expires_on
+    application_link && contact_email && publish_on && expires_on && expiry_time
+  end
+
+  def validate_expiry_time
```

Comment:
Some of these methods are private and some are public, be nice to be explicit about those which are private.

Reply:
Resolved this in e543ec7